gitbook-docs/rest_api.md

```diff
@@ -65,3 +65,9 @@ of the data under the `vessels` container in JSON format. Likewise, `GET
 /signalk/v1/api/vessels/urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d` should return data for one specific
 vessel. In other words, the full Signal K data model SHOULD be traversable by any client making GET requests to an
 arbitrary depth.
+
+## History snapshot retrieval
+
+A server MAY support retrieving historical data. The client specifies the requested timestamp with request parameter `time`, for example `https://localhost:3443/signalk/v1/api/vessels/self?time=2018-08-24T15:19:09Z`. The server will attempt to create the request part of the full model at the requested time.
```

Comment:
I have noticed Rob42 mentioning that some server support a `*` in the path. I am not sure we define this anywhere in the spec, so I assume the request here is for one context only and the response is an object, not a list.

Reply:
> The server will attempt to create the request part of the full model at the requested time.

Considering that we have multiple server implementation, maybe we could be more specific:
 - The server will reconstruct the SignalK full state as it would have been served at this time.
 - All the values returned by the server will have a timestamp equal or small to the requested snapshot time.

Reply:
You are right, the word 'reconstruct' is probably enough here.

The other way I thought it could be interpreted is: "the server will grab all the 'updates' with the requested timestamp and serve them", which could have meant that if you asked for data at 10:00:00 but the gps update with position and sog was at 9:59:59.800 you would not get it.